### ISSUE TITLE
Is it possible to emulate a right mouse click?

### ISSUE THREAD

**Issue by guljam** [gdh1995/vimium-c#961]
I often download images from Twitter and
I understand that firefox extensions cannot save proper image files due to security issues. Unable to save with an extension such as <image name>.png
(LinkHints.activateDownloadImage downloads the image but cannot save it with the correct name)

That is, I want to emulate a right-click input on the link and manually open the download window to save it.

Is this possible?

Firefox version 114.0.2 
build ID 	20230619081400

windows 11 22H2 build 22621.1848

**Comment by gdh1995** [comment]
No, extensions can not trigger a standard context menu (which is usually triggered by a right click).

If `LinkHints.activateDownloadImage` does succeed in access images and let you download it, then as far as I can image, you may give the image a new name in a "Save to" dialog.

If you want to download and directly save to your computer with a proper name, without showing a dialog, then sorry it's impossible for extensions.

**Comment by elig0n** [comment]
You can use hover node `LinkHints.activateHover` and after link selection use your keyboard's context menu button (next to the right windows key usually)
